lib/internal/bootstrap_node.js

```diff
@@ -246,6 +243,42 @@
     perf.markMilestone(NODE_PERFORMANCE_MILESTONE_BOOTSTRAP_COMPLETE);
   }
 
+  {
+    const bindingObj = Object.create(null);
+
+    const getBinding = process.binding;
+    process.binding = function binding(module) {
+      module = String(module);
+      if (typeof bindingObj[module] === 'object')
+        return bindingObj[module];
+      bindingObj[module] = getBinding(module);
+      return bindingObj[module];
+    };
+
+    const getLinkedBinding = process._linkedBinding;
+    process._linkedBinding = function _linkedBinding(module) {
+      module = String(module);
+      if (typeof bindingObj[module] === 'object')
+        return bindingObj[module];
+      bindingObj[module] = getLinkedBinding(module);
+      return bindingObj[module];
```

Comment:
Minor anti-pattern, looking up twice.  Do this instead:
```js
let mod = bindingObj[module];
if (typeof mod !== 'object')
  mod = bindingObj[module] = getBinding(module);
return mod;
```

Reply:
Good catch, don't know what I was thinking :) Updated.